pyqtgraph/graphicsItems/PlotItem/PlotItem.py

```diff
@@ -260,8 +314,85 @@ def __init__(self, parent=None, name=None, labels=None, title=None, viewBox=None
+        updateAxisCallback : Callable[[AxisItem, bool, ...], None], optional
+            Function to call on every axis. First argument is the AxisItem being modified, second argument is whether
+            this particular transform is currently enabled. E.g.::
+            lambda axis, enabled: axis.setLogMode(y=enabled)
+
+            If a `params` arg is passed to `addDataTransformOption`, this function can expect to have the parameters
+            therein described to be passed as named arguments.
+        params : dict, optional
+            Pass in a parameter tree config (see: ParameterTree) to generate an additional UI for said parameters. These
+            parameters will then be passed into all invocations of the `dataTransform` and `updateAxisCallback`
+            functions as named parameters. Only flat, non-nested parameters are currently supported.
+        order : int, optional
+            Specify the order in which transforms should be applied. Default value is 50, which is before all the
+            default transforms. Should be in the 0-100 range. Ties resolve in the order they're added.
+
+        Returns
+        -------
+        None
+
+        See Also
+        --------
+        PlotItem.addDefaultDataTransformOption
+        """
+        if name in self._transforms:
+            raise ValueError(f"A transform with name '{name}' is already present.")
+        self._transforms[name] = {
+            "dataTransform": dataTransform,
+            "updateAxisCallback": updateAxisCallback,
+            "params": params,
+            "order": order,
+        }
+        row = self.ctrl.gridLayout.rowCount()
+        check = QtWidgets.QCheckBox(self.ctrl.transformGroup)
+        self._transforms[name]["checkbox"] = check
+        check.setObjectName(name)
+        check.setText(name)
+        self.ctrl.gridLayout.addWidget(check, row, 0, 2, 1)
+        if params is not None:
+            paramsParams = Parameter.create(name=f"{name} params", type="group", children=params)
+            paramsParams.setObjectName(name)
+            paramsUi = ParameterTree(showHeader=False)
+            paramsUi.addParameters(paramsParams, showTop=False)
+            paramsUi.setFrameStyle(QtWidgets.QFrame.Shadow.Raised | QtWidgets.QFrame.Shape.Panel)
+            self.ctrl.gridLayout.addWidget(paramsUi, row, 1, 1, 1)
+            paramsParams.sigTreeStateChanged.connect(self._handleDataTransformParamsUpdate)
+            self._transforms[name]["paramsParams"] = paramsParams
+            self._transforms[name]["paramsUi"] = paramsUi
+            self._transforms[name]["params"] = params
+            paramsParams.sigTreeStateChanged.emit(paramsParams, None)  # registers with all the data items
+        check.toggled.connect(self._handleDataTransformChecked)
+        check.toggled.emit(False)  # registers with all the data items and axes
+
+    # :MC: removeDataTransformOption was left unimplemented. Here are the concerns it will need to cover:
```

Comment:
This is the most glaring shortcoming of this PR; it feels like removing transform options from a particular plot is something users are going to want to do. That said, the notes about why it wasn't included do make for a good starting point if someone decides to add this as a feature later.